packages/financial-templates-lib/src/clients/OptimisticOracleClient.js

```diff
@@ -0,0 +1,118 @@
+// A thick client for getting information about an OptimisticOracle. Used to get price requests and
+// proposals, which can be disputed and settled.
+const { OptimisticOracleRequestStatesEnum } = require("@uma/common");
+const Promise = require("bluebird");
+
+class OptimisticOracleClient {
+  /**
+   * @notice Constructs new OptimisticOracleClient.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} oracleAbi OptimisticOracle truffle ABI object to create a contract instance.
+   * @param {Object} web3 Provider from Truffle instance to connect to Ethereum network.
+   * @param {String} oracleAddress Ethereum address of the OptimisticOracle contract deployed on the current network.
+   * @return None or throws an Error.
+   */
+  constructor(logger, oracleAbi, web3, oracleAddress) {
+    this.logger = logger;
+    this.web3 = web3;
+
+    // Oracle contract
+    this.oracle = new web3.eth.Contract(oracleAbi, oracleAddress);
+    this.oracleAddress = oracleAddress;
+
+    // Oracle Data structures & values to enable synchronous returns of the state seen by the client.
+    this.priceRequests = [];
+    this.priceProposals = [];
+
+    // Store the last on-chain time the clients were updated to inform price request information.
+    this.lastUpdateTimestamp = 0;
+
+    // Helper functions from web3.
+    this.hexToUtf8 = this.web3.utils.hexToUtf8;
+  }
+
+  // Returns an array of Price Request objects that have no proposals yet.
+  getAllPriceRequests() {
+    return this.priceRequests;
+  }
+
+  // Returns an array of Price Request objects that someone has proposed a price for.
+  getAllPriceProposals() {
```

Comment:
Will the bot try to settle its own proposals? If so, should we add a similar method for "settleable" requests?

Reply:
Yeah that's an option, future PR work as I just want to get some resolution about how we query price requests in general.

Reply:
Sounds good! Just wanted to raise the issue.